core/src/main/java/net/opentsdb/pools/BlockingQueueArrayObjectPool.java

```diff
@@ -0,0 +1,173 @@
+          + "'ArrayObjectPoolAllocator'");
+    }
+    pool = new ArrayBlockingQueue<PooledObject>(count);
+    for (int i = 0; i < count; i++) {
+      final Object obj = config.allocator().allocate();
+      pool.offer(new LocalPooled(obj, true));
+    }
+    tsdb.getMaintenanceTimer().newTimeout(this, 60, TimeUnit.SECONDS);
+    if (LOG.isDebugEnabled()) {
+      LOG.debug("Instantiated object pool with " + count + " entries for " 
+          + config.id());
+    }
+  }
+  
+  @Override
+  public PooledObject claim() {
+    PooledObject obj = pool.poll();
+    if (obj == null) {
+      tsdb.getStatsCollector().incrementCounter("objectpool.claim.miss", 
+          "pool", config.id());
+      return new LocalPooled(config.allocator().allocate(), false);
+    } else {
+      tsdb.getStatsCollector().incrementCounter("objectpool.claim.success", 
+          "pool", config.id());
+      return obj;
+    }
+  }
+
+  @Override
+  public PooledObject claim(final long time, final ChronoUnit unit) {
+    // TODO - time it and retry
+    return claim();
+  }
+
+  @Override
+  public PooledObject claim(final int length) {
+    if (length > ((ArrayObjectPoolAllocator) config.allocator()).pooledLength()) {
+      tsdb.getStatsCollector().incrementCounter("objectpool.claim.largeArray", 
+          "pool", config.id());
+      return new LocalPooled(((ArrayObjectPoolAllocator) 
+          config.allocator()).allocate(length), false);
+    }
+    return claim();
+  }
+
+  @Override
+  public PooledObject claim(final int length, 
+                            final long time, 
+                            final ChronoUnit unit) {
+    // TODO - time it and retry
+    return claim(length);
+  };
+  
+  @Override
+  public String id() {
+    return config.id();
+  }
+
+  @Override
+  public Deferred<Object> shutdown() {
+    return Deferred.fromResult(null);
+  }
+
+  public void run(final Timeout timeout) {
+    try {
+      tsdb.getStatsCollector().setGauge("objectpool.available", pool.size(),
+          "pool", config.id());
+    } catch (Throwable t) {
+      LOG.error("Failed to run the metric timer task for " + config.id(), t);
+    } finally {
+      tsdb.getMaintenanceTimer().newTimeout(this, 60, TimeUnit.SECONDS);
+    }
+  }
+  
+  /**
+   * Super simple wrapper.
+   */
+  private class LocalPooled implements PooledObject {
+    final Object obj;
+    final boolean was_pooled;
+    
+    protected LocalPooled(final Object obj, final boolean was_pooled) {
+      this.obj = obj;
+      if (obj instanceof CloseablePooledObject) {
+        ((CloseablePooledObject) obj).setPooledObject(this);
+      }
+      this.was_pooled = was_pooled;
+    }
+    
+    @Override
+    public Object object() {
+      return obj;
+    }
+
+    @Override
+    public void release() {
+      if (was_pooled) {
+        if (!pool.offer(this)) {
+          LOG.warn("Failed to return a pooled object to the pool for " + config.id());
```

Comment:
should decrement the count

Reply:
oh and I don't actually have a count here to decrement. The config does have it but that's just an initialization time value. I can add a "size" getter if we need it.